### ISSUE TITLE
RequireJS throws a "mismatch" error

### ISSUE THREAD

**Issue by nwhittaker** [typekit/webfontloader#278]
If a project is already using RequireJS, Web Font Loader detects it and creates an anonymous module. In cases where Web Font Loader is loaded statically via a `script` tag (e.g. Google Charts API), and not dynamically via AMD, it causes Require to throw a "Mismatched anonymous define() module" error. This error halts further JavaScript execution.

Please see [plunker for example](http://plnkr.co/edit/QbH3jkBan8ItyyU0Yjm0?p=preview).

**Comment by kynatro** [comment]
I am also having this issue, but it only recently appeared. This only recently started happening in our application. Screenshot of the offending line: http://take.ms/3Iduw.

**Comment by bramstein** [comment]
Mm, this is probably caused by the UMD wrapper that was introduced when the Web Font Loader officially started supporting modules. I'm not that familiar with RequireJS, but it looks like creating a named module would create some other problems. Any suggestions?

**Comment by pre2612** [comment]
Is there a solution for this?
Having same issues when trying to use it in Magento 2 project?

Any help would be appreciated!

**Comment by ioweb-gr** [comment]
@pre2612 Did you get anywhere with this?

**Comment by pre2612** [comment]
Yes it worked there was a solution found online which helped resolved it

**Comment by ioweb-gr** [comment]
@pre2612 Nice. Could you let us know what it was that helped you?

**Comment by Deepak-Bulani** [comment]
@nwhittaker , Did you got any success for this issue?
 I am still facing this issue and couldn't find any way out.

**Comment by ioweb-gr** [comment]
@Deepak-Bulani I faced a similar issue when trying to create my own library through webpack which would be published as an AMD module for use with requirejs. 

After reading a lot about it I ended up in this section of the documentation. 

https://requirejs.org/docs/errors.html#mismatch

Apparently if you pack your library in AMD format like this 

`define([], function(){})`

and include it via script it will always produce a `Mismatched anonymous define()`

Even if you require it via requirejs it will randomly present this error. Once it will work and once it will fail because of the async loading order. 

I ended up having to use a name in the `define` section of my module in order to make it work consistently even though it's recommended against this. 

Reading further you will notice that if you use the RequireJs optimizer which will bundle the modules, it will also define a name automatically for each one of the defined modules. 

So basically I think this issue is very relevant to what I faced and I haven't found a workaround with pure requirejs and my library other than naming the module itself.

**Comment by prdt3e** [comment]
https://github.com/requirejs/requirejs/issues/1479
